pkg/auth/ldaputil/ldapclient/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package ldapclient
+
+import (
+	"crypto/tls"
+
+	"github.com/go-ldap/ldap"
+)
+
+// Client knows how to interact with an LDAP server
+type Client interface {
```

Comment:
explain the purpose of the interface and describe where you pulled the methods from.